packages/kovan-faucets/src/ts/handler.ts

```diff
@@ -0,0 +1,91 @@
+import * as express from 'express';
+import * as _ from 'lodash';
+import * as Web3 from 'web3';
+import ProviderEngine = require('web3-provider-engine');
+import HookedWalletSubprovider = require('web3-provider-engine/subproviders/hooked-wallet');
+import NonceSubprovider = require('web3-provider-engine/subproviders/nonce-tracker');
+import RpcSubprovider = require('web3-provider-engine/subproviders/rpc');
+
+import {configs} from './configs';
+import {EtherRequestQueue} from './ether_request_queue';
+import {idManagement} from './id_management';
+import {utils} from './utils';
+import {ZRXRequestQueue} from './zrx_request_queue';
+
+// HACK: web3 leaks XMLHttpRequest into the global scope and causes requests to hang
+// because they are using the wrong XHR package.
+// Issue: https://github.com/trufflesuite/truffle-contract/issues/14
+delete (global as any).XMLHttpRequest;
```

Comment:
We can actually remove this hack by simply moving the `web3` import above after the subprovider imports. The subproviders packages look at this  global variable to decide if they are in a browser or node env. Web3 sets this global var :(